sale_revert_done/tests/test_sale_revert_done.py

```diff
@@ -0,0 +1,18 @@
+# coding: utf-8
+from odoo.tests.common import TransactionCase
+from odoo.exceptions import ValidationError
+
+
+class TestSaleRevertDone(TransactionCase):
+    def test_sale_revert_done(self):
+        order = self.env.ref('sale.sale_order_4').copy()
+        order.action_confirm()
+        self.assertEqual(order.state, 'sale')
+        order.action_done()
+        self.assertEqual(order.state, 'done')
+        order.action_revert_done()
+        self.assertEqual(order.state, 'sale')
+        with self.assertRaisesRegexp(ValidationError, 'not in state'):
```

Comment:
As a side note, I prefer to not check texts, only exception types, because they can change on the life cycle of the module or because the user is not in 'English' language. It's very weird to have an exception in these small pieces of code that is not the one we are looking for.

Reply:
Will the nitpicking ever stop!

Reply:
This is not blocking, just commenting between colleagues. You have already my approval.